src/NATS.Client.Core/NatsMsg.cs

```diff
@@ -141,4 +141,6 @@ private void CheckReplyPreconditions()
             throw new NatsException("unable to send reply; ReplyTo is empty");
         }
     }
+
+    public bool IsNoRespondersError => Headers?.Code == 503;
```

Comment:
I don't know if it makes sense to offer just 1 public `IsNoRespondersError` here.  If we do one, we should probably enumerate every single error code.  I don't know if using Go or Rust as examples is great.  Those languages don't have exceptions.

Reply:
Not a standard error very specific [message from the server](https://github.com/nats-io/nats-server/blob/d96b489963cca10981b45342f9fdb008c2f7bf88/server/client.go#L3922). Other errors should still be check from the headers. we can't rely throw exceptions at this level without destroying the subscription.

⚠️ we should also check that payload is empty as other clients are doing. We don't have a reliable way of doing that (so ignore it for this issue). I'll create a separate issue for that.